File: y2021/cpp/day1.cpp
#include <fstream>
#include <iostream>
using namespace std;


int main()
{
    ifstream inputFile ("../inputs/day_1");
    int prev;
    int curr;
    int result;
    inputFile >> prev;

    while(!inputFile.eof()){
        inputFile >> curr;
        if (curr > prev){
            result++;
        }
        prev = curr;
    }
    cout << result <<endl;
}

File: y2021/java/Day1.java
package y2021.java;
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public final class Day1 {
    public static void main(String[]args) throws IOException {
        BufferedReader br = new BufferedReader(new FileReader("y2021/inputs/day_1"));
        String line = br.readLine();
        int prev = Integer.parseInt(line);
        int curr;
        int result = 0;

        line = br.readLine();
        while (line != null)
        {
            curr = Integer.parseInt(line);
            if (curr > prev) {
                result += 1;
            }
            prev = curr;
            line = br.readLine();
        }
        System.out.println(result);
    }
}

